File: benchmarks/tasks/primes/espruino/impl.js

function is_prime(n) {
    if (n < 3) {
        return n > 1;
    } else {
        if ((n % 2 == 0) || (n % 3 == 0)) {
            return false;
        } else {
            let i = 5;
            while (i * i <= n) {
                if ((n % i == 0) || (n % (i + 2) == 0)) {
                    return false;
                }
                i += 6;
            }
            return true;

        }
    }
}

function main() {
    let sum = 0;
    let count = 0;
    for (let i = 1; sum < 133742; i++) {
        if (is_prime(i)){
            sum += i;
            count++;
        }
    }
    return count;
}



console.log("START")
for (let index = 0; index < 10; index++) {

    console.log(main())
}
console.log("DONE")

File: benchmarks/tasks/primes/wast/impl.c
int __attribute__((noinline)) is_prime(unsigned n) {
    if (n < 3) {
        return n > 1;
    } else {
        if ((n % 2 == 0) || (n % 3 == 0)) {
            return 0;
        } else {
            unsigned i = 5;
            while (i * i <= n) {
                if ((n % i == 0) || (n % (i + 2) == 0)) {
                    return 0;
                }
                i += 6;
            }
            return 1;
        }
    }
}

/*
 *
 * function is_prime(n)
    if n ≤ 3
       return n > 1
    else if n mod 2 = 0 or n mod 3 = 0
       return false
    let i ← 5
    while i * i ≤ n
       if n mod i = 0 or n mod (i + 2) = 0
           return false
       i ← i + 6
    return true
 * */
int main() {
    unsigned sum = 0;
    int count = 0;
    for (unsigned i = 1; sum < 13374242; i++) {
        if (is_prime(i)) {
            sum += i;
            count++;
        }
    }
    return count;
}
